pkg/generate/app/pipeline.go

```diff
@@ -341,21 +342,40 @@ func AddServices(objects Objects, firstPortOnly bool) Objects {
 	for _, o := range objects {
 		switch t := o.(type) {
 		case *deploy.DeploymentConfig:
-			ports := UniqueContainerToServicePorts(AllContainerPorts(t.Spec.Template.Spec.Containers...))
-			if len(ports) == 0 {
-				continue
+			svc := addServiceInternal(t.Spec.Template.Spec.Containers, t.ObjectMeta, t.Spec.Selector, firstPortOnly)
+			if svc != nil {
+				svcs = append(svcs, svc)
 			}
-			if firstPortOnly {
-				ports = ports[:1]
+		case *extensions.DaemonSet:
+			// Daemonsets have defaulting logic, if the selector is not set it will default to
+			// the labels on the pod template
+			selector := t.Spec.Template.Labels
+			if t.Spec.Selector != nil {
+				selector = t.Spec.Selector.MatchLabels
```

Comment:
Still seems weird to me that your service selector would not be the exact same selector that the DS uses to determine what pods it is managing.  If we create it based on pod labels the only weirdness that could happen is that a DS is still managing the pod but it no longer appears in the service.   I don't think we can run into a scenario where a pod not managed by the DS shows up in the service without some deliberate updates to the service.

Will change.